src/middlewared/middlewared/plugins/vm.py

```
@@ -776,72 +771,69 @@ def random_mac(self):
         Int('memory', required=True),
         Str('bootloader', enum=['UEFI', 'UEFI_CSM', 'GRUB']),
         Str('grubconfig', null=True),
-        List('devices', default=[], items=[Ref('vmdevice_create')]),
+        List('devices', default=[], items=[Patch('vmdevice_create', 'vmdevice_update', ('rm', {'name': 'vm'}))]),
         Bool('autostart', default=True),
         Str('time', enum=['LOCAL', 'UTC'], default='LOCAL'),
         register=True,
     ))
     async def do_create(self, data):
         """
-        Create a Virual Machine (VM).
+        Create a Virtual Machine (VM).
 
         `grubconfig` may either be a path for the grub.cfg file or the actual content
         of the file to be used with GRUB bootloader.
+
+        `devices` is a list of devices to add to the newly created Virtual Machine. If attaching a device fails for
+        some reason, VM would be destroyed and any resources allocated by the devices as well.
         """
 
         verrors = ValidationErrors()
         await self.__common_validation(verrors, 'vm_create', data)
         verrors.check()
 
         devices = data.pop('devices')
-        vm_id = None
-        created_zvols = []
-        try:
-            vm_id = await self.middleware.call('datastore.insert', 'vm.vm', data)
-
+        vm_id = await self.middleware.call('datastore.insert', 'vm.vm', data)
+        success = await self.safe_devices_updates(devices)
```

Why we dont raise the original traceback instead of asking to check logs?